packages/edit-navigation/src/components/header/save-button.js

```diff
@@ -19,6 +23,7 @@ export default function SaveButton( { navigationPost } ) {
 			isPrimary
 			onClick={ () => {
 				saveNavigationPost( navigationPost );
+				saveMenuName();
```

Comment:
I didn't anticipate this, but both these lines result in the menu being saved. `saveNavigationPost` saves the menu via the customizer API (because the REST API doesn't quite work yet for saving menu items).

I'd be worried about race conditions here.

This is some technical debt, so not really something caused by your PR.

I don't have a solution available right now, will have to have a think about it.

Reply:
what is the edge case when the race condition could really happen?

Reply:
I'm worried that one request might overwrite the other, since both API requests can modify the same database record.

I think the easiest solution right now would be to move the `saveEditedEntityRecord` dispatch inside the `saveNavigationPost` action to make sure the requests don't happen in parallel. Should be able to dispatch it with code like this, where the `yield` will wait for the request to complete before triggering the second request (the call to `batchSave`):
```js
yield dispatch(
	'core',
	'saveEditedEntityRecord',
	'root',
	'menu',
	menuId
);
```

This makes sense actually, as it means the on-screen notices that are implemented in that action will only show when both requests have finished.

Reply:
Some further background. The way saving menus currently works is a temporary thing. Currently the `batchSave` function has been patched to use an endpoint that is currently used by the Customize > Menus screen. But that isn't part of the REST API, and not something we want to support long into the future for the navigation editor.

Ideally the whole system would use the WordPress REST APIs and the core-data package's entities. There are two things that need to happen:
- Add a way to reorder menu items using the REST API (https://github.com/WordPress/gutenberg/issues/25093)
- Update the navigation editor to use the core-data package's newly introduced batch save API (from https://github.com/WordPress/gutenberg/pull/28210) for saving menus and menu items.